actions/wait-for-deployment/entrypoint.sh

```diff
@@ -2,6 +2,21 @@
 
 set -eu
 
+#----------- [API Deployment Only] Wait for Integration API Pod to spin back up with correct version --------------
+if [ "${DEPLOYMENT_TYPE}" = "api" ]; then
+    while true; do
+        echo "Checking ${API_URL}"
+        CURRENT_GITHASH=$(curl -s -X GET "${API_URL}/configuration" -H "accept: application/json" | jq -R '. | try fromjson catch {"gitHash":"failedToContact"}' | jq -r '.gitHash')
+        if [ "${DESIRED_GITHASH}" = "${CURRENT_GITHASH}" ]; then
```

Comment:
Something to keep in mind is if we use this action for a deployment that has more than one pod, it's possible that only one of the pods is on the desired version while the others are still updating. Integration and perf only have one API pod though so it won't be an issue there!

Reply:
That's a great point. I'm not sure how we could handle that situation.  

Given our specific situation, I think we're okay, but you're totally right - for a more general "wait for deployment" we'd need to solve this issue.
1. For perf, we do run multiple pods, but we delete them all first. So, this action basically serves as a wait for them to spin back up. If they're up, they should be on the right version.
2. For integration, we only run one pod and as you mentioned, that shouldn't be an issue.